homeassistant/components/netgear_lte/__init__.py

```diff
@@ -177,102 +185,155 @@ def get_modem_data(self, config):
 
 async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     """Set up Netgear LTE component."""
-    if DATA_KEY not in hass.data:
-        websession = async_create_clientsession(
-            hass, cookie_jar=aiohttp.CookieJar(unsafe=True)
-        )
-        hass.data[DATA_KEY] = LTEData(websession)
-
-        async def service_handler(service: ServiceCall) -> None:
-            """Apply a service."""
-            host = service.data.get(ATTR_HOST)
-            conf = {CONF_HOST: host}
-            modem_data = hass.data[DATA_KEY].get_modem_data(conf)
-
-            if not modem_data:
-                _LOGGER.error("%s: host %s unavailable", service.service, host)
-                return
-
-            if service.service == SERVICE_DELETE_SMS:
-                for sms_id in service.data[ATTR_SMS_ID]:
-                    await modem_data.modem.delete_sms(sms_id)
-            elif service.service == SERVICE_SET_OPTION:
-                if failover := service.data.get(ATTR_FAILOVER):
-                    await modem_data.modem.set_failover_mode(failover)
-                if autoconnect := service.data.get(ATTR_AUTOCONNECT):
-                    await modem_data.modem.set_autoconnect_mode(autoconnect)
-            elif service.service == SERVICE_CONNECT_LTE:
-                await modem_data.modem.connect_lte()
-            elif service.service == SERVICE_DISCONNECT_LTE:
-                await modem_data.modem.disconnect_lte()
-
-        service_schemas = {
-            SERVICE_DELETE_SMS: DELETE_SMS_SCHEMA,
-            SERVICE_SET_OPTION: SET_OPTION_SCHEMA,
-            SERVICE_CONNECT_LTE: CONNECT_LTE_SCHEMA,
-            SERVICE_DISCONNECT_LTE: DISCONNECT_LTE_SCHEMA,
-        }
+    hass.data[DATA_HASS_CONFIG] = config
```

Comment:
I think you should move this under the `DOMAIN` key, either as a dict with sub-items or possibly as a dataclass.

Reply:
This constants resolves to `"netgear_lte_hass_config"`. Much will eventually be moved to coordinator and leave just the `notify` service using it.